src/Avalonia.X11/X11TrayIconImpl.cs

```diff
@@ -0,0 +1,381 @@
+
+            _backingProperties = new StatusNotifierItemProperties
+            {
+                Menu = dbusmenuPath, // Needs a dbus menu somehow
+                ToolTip = new ToolTip("")
+            };
+
+            InvalidateAll();
+        }
+
+        public async Task ContextMenuAsync(int X, int Y)
+        {
+        }
+
+        public async Task ActivateAsync(int X, int Y)
+        {
+            ActivationDelegate?.Invoke();
+        }
+
+        public async Task SecondaryActivateAsync(int X, int Y)
+        {
+        }
+
+        public async Task ScrollAsync(int Delta, string Orientation)
+        {
+        }
+
+        public void InvalidateAll()
+        {
+            OnTitleChanged?.Invoke();
+            OnIconChanged?.Invoke();
+            OnOverlayIconChanged?.Invoke();
+            OnAttentionIconChanged?.Invoke();
+            OnTooltipChanged?.Invoke();
+        }
+
+        public async Task<IDisposable> WatchNewTitleAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnTitleChanged += handler;
+            return Disposable.Create(() => OnTitleChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewIconAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnIconChanged += handler;
+            return Disposable.Create(() => OnIconChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewAttentionIconAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnAttentionIconChanged += handler;
+            return Disposable.Create(() => OnAttentionIconChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewOverlayIconAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnOverlayIconChanged += handler;
+            return Disposable.Create(() => OnOverlayIconChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewToolTipAsync(Action handler, Action<Exception> onError = null)
+        {
+            OnTooltipChanged += handler;
+            return Disposable.Create(() => OnTooltipChanged -= handler);
+        }
+
+        public async Task<IDisposable> WatchNewStatusAsync(Action<string> handler, Action<Exception> onError = null)
+        {
+            NewStatusAsync += handler;
+            return Disposable.Create(() => NewStatusAsync -= handler);
+        }
+
+        public async Task<object> GetAsync(string prop)
+        {
+            return null;
+        }
+
+        public async Task<StatusNotifierItemProperties> GetAllAsync()
+        {
+            return _backingProperties;
+        }
+
+        public Task SetAsync(string prop, object val) => Task.CompletedTask;
+
+        public async Task<IDisposable> WatchPropertiesAsync(Action<PropertyChanges> handler)
+        {
+            OnPropertyChange += handler;
+            return Disposable.Create(() => OnPropertyChange -= handler);
+        }
```

Comment:
lots of warnings here. Async is not needed in those places.

Reply:
fixed

Reply:
should be resolved now.